include/arch/arm/aarch32/irq.h

```diff
@@ -235,6 +235,15 @@ extern void _isr_wrapper(void);
 
 #endif /* _ASMLANGUAGE */
 
+#ifdef CONFIG_ARMV6_M_ARMV8_M_BASELINE
+#define NVIC_NODEID DT_INST(0, arm_v6m_nvic)
```

Comment:
My grep says DT_NODELABEL(nvic) would work just as well in all cases, and lets the .dtsi files handle future SoCs. Why not use that?

Reply:
I leave @ioannisg to comment, but I think this header is "private" to arm based on where it is.

Reply:
i think so, it should be private to arm